src/Diagnostics.RuntimeHost/Middleware/DiagnosticsRequestMiddleware.cs

```diff
@@ -67,11 +67,7 @@ private void BeginRequest_Handle(HttpContext httpContext)
 
             httpContext.RequestAborted = cTokenSource.Token;
             httpContext.Request.Headers.TryGetValue(HeaderConstants.RequestIdHeaderName, out StringValues values);
-            Dictionary<string, string> queryStringValues = new Dictionary<string, string>();
-            foreach (var query in httpContext.Request.Query)
-            {
-                queryStringValues.Add(query.Key.ToLower(), query.Value.FirstOrDefault());
-            }
+            var queryStringValues = httpContext.Request.Query.ToDictionary(query => query.Key.ToLower(), query => query.Value.FirstOrDefault());
```

Comment:
Can the `Query` part of Request be `null`? In that case `ToDictionary` will give `NullReferenceException`.